python/mxnet/ndarray/numpy/_op.py

```diff
@@ -5308,3 +5307,57 @@ def nan_to_num(x, copy=True, nan=0.0, posinf=None, neginf=None, **kwargs):
         return _npi.nan_to_num(x, copy=copy, nan=nan, posinf=posinf, neginf=neginf, out=None)
     else:
         raise TypeError('type {} not supported'.format(str(type(x))))
+
+
+@set_module('mxnet.ndarray.numpy')
+def diagflat(arr, k=0):
+    """
+    Create a two-dimensional array with the flattened input as a diagonal.
+    Parameters: 
+    
+    v : array_like
```

Comment:
```suggestion
    arr : ndarray
```
or you can fall back to numpy when arr is not ndarray type